src/api/http.ts

```diff
@@ -0,0 +1,83 @@
+import { API_TOKEN, BASE_API_URL } from './constants';
+
+type RequestMethod = 'GET' | 'POST' | 'PUT' | 'PATCH' | 'DELETE';
+
+const request = async <TBody, TReturn>(
+  method: RequestMethod,
+  url: string,
+  data?: TBody,
+  headerName?: string
+): Promise<TReturn> => {
+  const response = await fetch(url, {
+    method,
+    body: data ? JSON.stringify(data) : undefined,
+    headers: {
+      'Content-Type': 'application/json',
+      Authorization: `Bearer ${API_TOKEN}`,
+    },
+  });
+
+  if (!response.ok) {
+    throw new Error('Network response was not ok');
+  }
+
+  let headerValue;
+  if (headerName) {
+    headerValue = response.headers.get(headerName ?? '');
+  }
+
+  const body = await response.text();
+  try {
+    if (headerName && headerValue) {
+      return {
+        [headerName]: headerValue,
+        data: JSON.parse(body),
+      } as TReturn;
+    }
+    return JSON.parse(body);
+  } catch {
+    return body as TReturn;
+  }
+};
+
+export const getRequest = async <
```

Comment:
Can it be simplified?

```suggestion
export const getRequest = async <TData>(
  url: string,
  queryParams?: Record<string, string> = {},
  headerName?: string
): Promise<TData> => {
  const urlWithQuery = new URL(`${BASE_API_URL}${url}`).toString() + new URLSearchParams(queryParams);
  return await request('GET', urlWithQuery, undefined, headerName);
};
```